shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -554,7 +554,34 @@ private void copyFileToLocal(AlluxioURI srcPath, AlluxioURI dstPath)
     File dstFile = new File(dstPath.getPath());
     String randomSuffix =
         String.format(".%s_copyToLocal_", RandomStringUtils.randomAlphanumeric(8));
-    File tmpDst = new File(dstFile.getAbsolutePath() + randomSuffix);
+    File tmpDst;
+    File outputFile;
+    String dstAbsolute = dstFile.getAbsolutePath();
+    String fileName;
+    if (dstPath.getPath().equals("")) {
+      // current directory
+      if (dstPath.getAuthority().equals(".")) {
+        fileName = srcPath.getName();
+      } else {
+        fileName = dstPath.getAuthority();
+      }
+    } else {
+      fileName = dstPath.getName();
+    }
+
+    if (new File(dstAbsolute).exists()) {
```

Comment:
when the destination is a file, we can treat it the same as if it doesn't exist, so we only need two cases based on `isFile()`